SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosStatusUpdateHandler.java

```diff
@@ -344,6 +346,35 @@ private StatusUpdateResult unsafeProcessStatusUpdate(
           taskIdObj
         );
         return StatusUpdateResult.KILL_TASK;
+      } else if (
+        maybeTaskHistory.isPresent() &&
+        status.getReason() == Reason.REASON_AGENT_REREGISTERED
+      ) {
+        boolean lbRemovalStarted = maybeTaskHistory
+          .get()
+          .getLoadBalancerUpdates()
+          .stream()
+          .anyMatch(
+            lbUpdate ->
+              lbUpdate.getMethod() == LoadBalancerMethod.CANCEL ||
+              lbUpdate.getMethod() == LoadBalancerMethod.DELETE
+          );
+        if (lbRemovalStarted) {
+          taskManager.createTaskCleanup(
+            new SingularityTaskCleanup(
+              null,
+              TaskCleanupType.PRIORITY_KILL,
```

Comment:
priority kill is an inta-kill, probably too harsh. We can have this shut down gracefully similar to when we replace it with a bounce. Could easily fall under `DECOMISSIONING` ?

Reply:
forget this actually, saw your follow up question below. if we return a `StatusUpdateResult.KILL_TASK` in the end, that will take care of it all for us

Reply:
Though not sure if it will take care of the cleaning + pending request. Would have to double check that. If not we'll need to stick to the cleanup being created + returning DONE

Reply:
returning the `KILL_TASK` will put a timeout of 1 minute for graceful termination on the task, so I think we want to go with the task cleanup + `DONE` here

Reply:
👍